tutor/src/components/my-courses/index.jsx

```diff
@@ -1,23 +1,56 @@
 import React from 'react';
 import { Redirect } from 'react-router-dom';
-import { computed } from 'mobx';
-import { observer } from 'mobx-react';
+import { computed, action } from 'mobx';
+import { observer, inject } from 'mobx-react';
 import User from '../../models/user';
 import Router from '../../helpers/router';
 import Courses from '../../models/courses-map';
 import EmptyCourses from './empty';
 import TourRegion from '../tours/region';
 import PendingVerification from './pending-verification';
+import TourContext from '../../models/tour/context';
+import { includes } from 'lodash';
 
 import { MyCoursesPast, MyCoursesCurrent, MyCoursesPreview } from './listings';
 
+@inject((allStores, props) => ({
+  tourContext: ( props.tourContext || allStores.tourContext ),
+}))
 @observer
 export default class MyCourses extends React.PureComponent {
 
+  static propTypes = {
+    tourContext: React.PropTypes.instanceOf(TourContext),
+  }
+
+  @computed get stubbornTour() {
+    return this.props.tourContext.eligibleTours.find((tour) => (
+        includes([
+          'explore-a-preview',
+          'create-a-course',
+        ], tour.id)
+      ));
+  }
+
   componentWillMount() {
     User.logEvent({ category: 'onboarding', code: 'arrived_my_courses' });
   }
 
+  componentWillReceiveProps(nextProps) {
+    this.forceTours();
+  }
+
+  @action.bound
+  forceTours() {
+    if (!this.props.tourContext.tour) {
+      const { stubbornTour } = this;
+
+      if (stubbornTour) {
+        stubbornTour.isEnabled = true;
+      }
+    }
+  }
+
   @computed get firstCourse() {
```

Comment:
Hm, what're you trying to do here?  It seems like there's a tour that's refusing to display?